gobblin-data-management/src/main/java/gobblin/data/management/convertion/hive/TableLevelWatermaker.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.data.management.convertion.hive;
+
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.hive.ql.metadata.Partition;
+import org.apache.hadoop.hive.ql.metadata.Table;
+
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+
+import gobblin.configuration.SourceState;
+import gobblin.configuration.State;
+import gobblin.configuration.WorkUnitState;
+import gobblin.source.extractor.Watermark;
+
+/**
+ * A {@link HiveSourceWatermarker} that manages {@link Watermark} at a per {@link Table} basis.
+ * One {@link Watermark} per table exists.
+ * All {@link Partition}s of a {@link Table} have the same {@link Watermark}.
+ */
+public class TableLevelWatermaker implements HiveSourceWatermarker {
+  public static final Gson GSON = new Gson();
+
+  // Table complete name db@tb - list of previous workunitState
+  private Map<String, LongWatermark> tableWatermarks;
+
+  public TableLevelWatermaker(State state) {
+    this.tableWatermarks = Maps.newHashMap();
+
+    if (state instanceof SourceState) {
```

Comment:
I think this is always true. You should check whether state is instanceof `DatasetState`. If so, skip the block.